zjit/src/hir.rs

```diff
@@ -1719,7 +1728,7 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
     let mut queue = std::collections::VecDeque::new();
     let mut entry_state = FrameState::new(iseq);
     for idx in 0..num_locals(iseq) {
-        if idx < num_lead_params(iseq) {
+        if idx < unsafe { get_iseq_body_param_size(iseq) }.as_usize() {
```

Comment:
Now that we apparently handle all cases (?) can you please add a comment explaining how different parameter types get turned into HIR? Positionals seem the most "obvious" but other stuff seems like it would take a logical leap

Reply:
I wrote one, but I'm not sure if it's the kind you're looking for. I think other than the hidden param for keyword params, the count is pretty much always the size of the comma separated parameter list in the source text.

Let me know if something is unclear.